### PR TITLE
Create separate module for API structs

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: .github/workflows/build-mariadb-operator.yaml [hunk 1]
```diff
@@ -91,7 +91,7 @@ jobs:
       uses: redhat-actions/openshift-tools-installer@v1
       with:
         source: github
-        operator-sdk: '1.18.1'
+        operator-sdk: '1.22.0'
```
Comment (stuggi) [inline .github/workflows/build-mariadb-operator.yaml:94]: 1.22.1?

File: .github/workflows/build-mariadb-operator.yaml [hunk 2]
```diff
@@ -82,7 +82,7 @@ jobs:
     - name: Install Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17.x
+        go-version: 1.18.0
```
Comment (stuggi) [inline .github/workflows/build-mariadb-operator.yaml:85]: 1.18.4?

File: .github/workflows/golangci-lint.yaml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.17.8
+          go-version: 1.18.0
```
Comment (stuggi) [inline .github/workflows/golangci-lint.yaml:35]: current lastest 1.18.4 ?

File: .github/workflows/build-mariadb-operator.yaml (hunk 2 above)
Comment (dprince) [reply .github/workflows/build-mariadb-operator.yaml:85]: I'm not sure setup-go always supports the latest go version. But lets try it.

Comment (dprince) [reply .github/workflows/build-mariadb-operator.yaml]: I think we might should just make all of these 1.18.x like what was there before.

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**